ml-algorithms/src/main/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactory.java

```diff
@@ -8,79 +8,55 @@
 import java.util.Arrays;
-
-import org.apache.commons.lang3.math.NumberUtils;
-import org.apache.http.HttpHost;
-import org.apache.http.HttpRequest;
-import org.apache.http.HttpResponse;
-import org.apache.http.conn.UnsupportedSchemeException;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClientBuilder;
-import org.apache.http.impl.client.LaxRedirectStrategy;
-import org.apache.http.impl.conn.DefaultSchemePortResolver;
-import org.apache.http.protocol.HttpContext;
-
-import com.google.common.annotations.VisibleForTesting;
+import java.util.Locale;
 
 import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
 
 @Log4j2
 public class MLHttpClientFactory {
 
-    public static CloseableHttpClient getCloseableHttpClient() {
-        return createHttpClient();
-    }
-
-    private static CloseableHttpClient createHttpClient() {
-        HttpClientBuilder builder = HttpClientBuilder.create();
-
-        // Only allow HTTP and HTTPS schemes
-        builder.setSchemePortResolver(new DefaultSchemePortResolver() {
-            @Override
-            public int resolve(HttpHost host) throws UnsupportedSchemeException {
-                validateSchemaAndPort(host);
-                return super.resolve(host);
-            }
-        });
-
-        builder.setDnsResolver(MLHttpClientFactory::validateIp);
-
-        builder.setRedirectStrategy(new LaxRedirectStrategy() {
-            @Override
-            public boolean isRedirected(HttpRequest request, HttpResponse response, HttpContext context) {
-                // Do not follow redirects
-                return false;
-            }
-        });
-        return builder.build();
+    public static SdkAsyncHttpClient getAsyncHttpClient() {
+        try {
+            return AccessController
+                .doPrivileged(
+                    (PrivilegedExceptionAction<SdkAsyncHttpClient>) () -> NettyNioAsyncHttpClient.builder().maxConcurrency(100).build()
+                );
+        } catch (PrivilegedActionException e) {
+            return null;
+        }
     }
 
-    @VisibleForTesting
-    protected static void validateSchemaAndPort(HttpHost host) {
-        String scheme = host.getSchemeName();
-        if ("http".equalsIgnoreCase(scheme) || "https".equalsIgnoreCase(scheme)) {
-            String[] hostNamePort = host.getHostName().split(":");
-            if (hostNamePort.length > 1 && NumberUtils.isDigits(hostNamePort[1])) {
-                int port = Integer.parseInt(hostNamePort[1]);
-                if (port < 0 || port > 65536) {
-                    log.error("Remote inference port out of range: " + port);
-                    throw new IllegalArgumentException("Port out of range: " + port);
-                }
+    public static void validate(String protocol, String host, int port) throws UnknownHostException {
```

Comment:
can we give more meaningful name? Validating what?

Reply:
This validates a few things I don't think we need to explicitly specify everything in the method name(which can make the method name very long), when one's reading the code and saw the method and its parameters, one should be able to understand the things we're validating. I don't see any confusion or inconvenience this method name will bring.